src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -1340,6 +1393,100 @@ def validate_arguments(self):
             if scale_rule_type == "http" or scale_rule_type == "tcp":
                 raise InvalidArgumentValueError("--scale-rule-identity cannot be set when --scale-rule-type is 'http' or 'tcp'")
 
+    def check_assign_system_identity(self):
+        identity = self.get_argument_registry_identity()
+        system_sp = safe_get(self.containerapp_def, "identity", "principalId")
+
+        # create system service principalId
+        if system_sp is None and is_registry_msi_system(identity):
```

Comment:
in this condition, only when customer specify system as identity, we will create the system principalId? should we also create in default scenario?

Reply:
Yes, we did.
For the default behavior, in the `set_up_system_assigned_identity_as_default_if_using_acr`, it will set identity to `system`